src/components/CCPADialog.jsx

```diff
@@ -0,0 +1,150 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import {
+  ModalDialog, Form, useToggle, ActionRow, Button,
+} from '@edx/paragon';
+import Cookies from 'js-cookie';
+
+const CCPA_COOKIE_NAME = 'edx_do_not_sell';
+
+const CCPADialog = ({ dialogIsOpen, closeCallback, baseURL }) => {
+  const [isOpen, open, close] = useToggle(dialogIsOpen);
+
+  const getDoNotSellCookie = () => Cookies.get(CCPA_COOKIE_NAME) === 'true';
+  const [personalizationChecked, setPersonalizationChecked] = useState(() => !getDoNotSellCookie());
+
+  useEffect(() => {
+    if (dialogIsOpen) {
+      setPersonalizationChecked(!getDoNotSellCookie());
+      open();
+    }
+  }, [dialogIsOpen, open]);
+
+  useEffect(() => {
+    if (!isOpen) {
+      closeCallback();
+    }
+  }, [closeCallback, isOpen]);
+
+  const setDoNotSellCookie = (value) => {
+    const { host } = new URL(baseURL);
+    // Use global domain (`.edx.org`) without the subdomain
```

Comment:
I usually like to extract a helper function and unit test string manipulation like this, even just to make the requirements explicit. We don't have to handle "http://edx.org" right? My understanding is that all domains will have a subdomain (either `www` or another string).

Reply:
that's my understanding as well.  https://edx.org redirects to www